uuid-cluster-dns/README.md

```diff
@@ -0,0 +1,49 @@
+# Using UUIDs for cluster endpoint domains
+
+The current approach for building cluster domains is to include the workload cluster's ID as a subdomain of the management cluster it belongs to, e.g.:
+
+```
+https://api.j9yr2.k8s.gauss.eu-west-1.aws.gigantic.io
+```
+
+This approach has a few drawbacks:
+
+1. The workload cluster is tied to the management cluster, making migration messy
+2. Workload clusters cannot be "promoted" to management clusters as the domain would no longer make sense
+3. Several customers have their own domain set up, leading to inconsistency between MCs (e.g. `eu-central-1.aws.cps.customer.com` vs. `gauss.eu-west-1.aws.gigantic.io`)
+4. The provider (and region) is baked into the domain, effectively preventing multi-region (or multi-cloud)
+
+An alternative "flat" DNS structure has been proposed that removes the link between WC and MC. E.g.:
+
+```
+https://api.j9yr2.k8s.[customer name].gigantic.io
+```
+
+While this solves some of the above drawbacks (1, 2, 4) it introduces its own complications:
+
+1. The customer name (or codename) is exposed in the domain
+2. There is no enforced global uniqueness on cluster IDs so if a customer has multiple MCs there is a chance that two workload clusters have the same ID, leading to conflicting domains
+
+## Proposal
+
+The flat DNS structure can be improved to avoid the drawbacks above my leveraging a UUID for the cluster ID rather than a user-providable short string.
+
+E.g.
+
+```
+https://api.4c6aba87-061c-458d-91b6-62ad07979728.k8s.gigantic.io
```

Comment:
I'm not super opinionated on this so this isn't intended as opposition. There are nice technical/functional upsides of the UUID approach.

Just adding here - it's not very nice to look at or work with as a human. Granted, we try to minimize manual touches and wrap these details in CLI sugar, but in cases where someone has to type this manually, or where they encounter a UUID and have to figure out which cluster it belongs to, this isn't great since they're unordered and opaque by design. I also sometimes benefit from seeing some of those details (e.g. region) in the base domain. Maybe others do as well.

So I'm not opposing UUIDs. Most of the downsides could be mitigated with `kgs` features, so maybe it's worth simultaneously opening an issue for tracking usability improvements if this is adopted.

Reply:
I kinda agree with your points but also agree these should be dealt with through tooling and we shouldn't rely on a domain to give insight into specifics (as we already have customer-provided domains that don't follow that convention)

Reply:
Just for context. The customer domains are old kvm installations that will be replaced with new installations and then turned off. There is only one (the oldest) AWS installation that uses a customer provided domain. We've stopped doing this because it is very painful.